### ISSUE TITLE
Main thread id stays 0 

### ISSUE THREAD

**Issue by dogukankse** [Benedicht/BestHTTP-Issues#186]
Unity Version: 2022.3.20f1
BestHTTP Version: 3.0.6 from the Asset Store

When I ran the app from the editor, I am getting this log:

> {"tid":1,"div":"HTTPUpdateDelegator","msg":"Setup called Threading Mode: UnityUpdate, MainThreadId: 1","stack":"  at Shared.HTTPUpdateDelegator.Setup () [0x00025] in .\/Packages\/com.tivadar.best.http\/Runtime\/Shared\/HTTPUpdateDelegator.cs:164   at Shared.HTTPUpdateDelegator.Update () [0x0000d] in .\/Packages\/com.tivadar.best.http\/Runtime\/Shared\/HTTPUpdateDelegator.cs:260 ","ctxs":[],"t":638453447676352460,"ll":"Information","bh":1}

However, with the first request, I encountered this error:

> {"tid":1,"div":"HTTPRequest","msg":"State.Set(Initial => Queued) isn't called on the main thread(0)!","stack":"  at HTTPRequest.set_State (HTTPRequestStates value) [0x00012] in .\/Packages\/com.tivadar.best.http\/Runtime\/HTTP\/HTTPRequest.cs:257   at Hosts.Connections.RequestEventHelper.ProcessQueue () [0x003a4] in .\/Packages\/com.tivadar.best.http\/Runtime\/HTTP\/Hosts\/Connections\/RequestEvents.cs:407   at Shared.HTTPManager.OnUpdate () [0x0004a] in .\/Packages\/com.tivadar.best.http\/Runtime\/Shared\/HTTPManager.cs:283   at Shared.HTTPUpdateDelegator.CallOnUpdate () [0x00017] in .\/Packages\/com.tivadar.best.http\/Runtime\/Shared\/HTTPUpdateDelegator.cs:273   at Shared.HTTPUpdateDelegator.Update () [0x00021] in .\/Packages\/com.tivadar.best.http\/Runtime\/Shared\/HTTPUpdateDelegator.cs:263 ","ctx": [{"TypeName": "HTTPRequest", "Hash": "49872fa244ebc2c7fc2df9cbae99e33f"}],"t":638453447991868710,"ll":"Error","bh":1}

Upon checking HTTPUpdateDelegator.mainThreadId, I noticed that it was set to 1, which is Unity's main thread ID. However, HTTPUpdateDelegator.Instance.IsMainThread() always returned false because HTTPUpdateDelegator.mainThreadId was 0.

I tried calling HTTPManager.Setup() when the app launched, but it didn't make any difference. Whether I called it manually or let the system handle it, the result was the same.

**Comment by Benedicht** [comment]
That's quite interesting, as you wrote it's initialized to 1 but when checked it's already 0.

How you are sending out requests, are they sent on different threads maybe? Is it happening 100%? Could you make a repro-project?

Anyway, while the log is displayed as an error, it's harmless in this case and wouldn't cause any issues.

**Comment by dogukankse** [comment]
It's becoming more interesting because I rarely shut down my Mac. However, after restarting the device yesterday, I no longer encounter the error. I have made no changes to the code; it's still using the same codebase.

I create a request with `HTTPRequest httpReq = new HTTPRequest(new Uri(request.URL))`, add some streams with `MemoryStream` or `MultipartFormDataStream`, and send it. I've also added my logger to these places, which can print the current working thread. It's all the same with your `tid` as expected.

I suspect that when I upgraded the package from version `2.8.3` to `3.0.6`, some instances like `HTTPUpdateDelegator` from the previous version might not have been destroyed. However, this seems unlikely.

There is no problem now. We can close the issue, or if you want to investigate further, I'm happy to lend some help.
